### DISCUSSION TITLE
Require user to agree to 'terms and conditions' during registration

### DISCUSSION THREAD

**Discussion by danieliu456** [keycloak/keycloak#23653] [Feedback]
Feature: https://github.com/keycloak/keycloak/pull/7938
Documentation: https://www.keycloak.org/docs/latest/server_admin/index.html#proc-requiring-tac-agreement-at-registration_server_administration_guide

Is there any possibility to set user attributes when the user accepts terms and conditions with this new flow?
Or did I misunderstand something? Because I can't see any attributes on the user after testing this functionality?

Dedicated terms and conditions page
```java
    @Override
    public void processAction(RequiredActionContext context) {
        // Keycloak 21.0.0 changed the user attribute name from lowercase to uppercase
        // this change was reverted, but it is still possible some attributes created
        // in Keycloak 21.0.0 will be present in the database, we need to remove it too.
        // See https://github.com/keycloak/keycloak/issues/17277 for more details
        context.getUser().removeAttribute(USER_ATTRIBUTE.toUpperCase());

        if (context.getHttpRequest().getDecodedFormParameters().containsKey("cancel")) {
            context.getUser().removeAttribute(USER_ATTRIBUTE);
            context.failure();
            return;
        }

        context.getUser().setAttribute(USER_ATTRIBUTE, Arrays.asList(Integer.toString(Time.currentTime())));

        context.success();
    }
```

Check box on user registration page:
```
	@Override
	public void success(FormContext context) {

	}
```

**Comment by dzhiurga** [comment]
Hello, 

Any updates on this? It's very much needed.

**Comment by SangI762** [comment]
We noticed the same behaviour. Also it would help us a lot.

**Comment by cuberinooo** [comment]
Same here. Any news on this ?

**Comment by Guite** [comment]
Created an issue at https://github.com/keycloak/keycloak/issues/25891 about this.
